.dotnet/src/Custom/Chat/OpenAIChatModelFactory.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenAI.Chat;
+
+/// <summary> Model factory for models. </summary>
+public static partial class OpenAIChatModelFactory
+{
+    /// <summary> Initializes a new instance of <see cref="OpenAI.Chat.ChatCompletion"/>. </summary>
+    /// <returns> A new <see cref="OpenAI.Chat.ChatCompletion"/> instance for mocking. </returns>
+    public static ChatCompletion ChatCompletion(
```

Comment:
FYI: I believe ChatCompletion has new properties from the Structured Outputs feature, so you might need to do a few updates.

Reply:
Thanks for letting me know. I have updated the code with the new properties (`Refusal` and `RefusalTokenLogProbabilities`). Are these the ones you're talking about? I couldn't find others.

Reply:
Yes, thank you!